### ISSUE TITLE
Feature request: SmithyBuildConfig -> Model utility

### ISSUE THREAD

**Issue by kubukoz** [smithy-lang/smithy#2292]
There are a lot of tools that could be built on top of Smithy, which would use `smithy-build.json` as the source of information about a model.

Currently, in order to do so they would have to either:

- shell out to the CLI (which is costly, including but not only because of having to start another JVM), or
- duplicate some of its logic: this is what the LSP seems to be doing at the moment, just like [smithy4s](https://github.com/disneystreaming/smithy4s/) / [smithy-playground](https://github.com/kubukoz/smithy-playground/) (OSS projects I'm involved with), and some projects I'm involved in at work.

I think it would be beneficial for the ecosystem to have more out-of-the-box support for loading a build config into a model. Presumably, this would have to be part of `smithy-build`.

**Comment by kstich** [comment]
Are you looking for more than what's exposed in [the `SmithyBuildConfig` class](https://github.com/smithy-lang/smithy/blob/main/smithy-build/src/main/java/software/amazon/smithy/build/model/SmithyBuildConfig.java)? If so, can you elaborate on what differences you're looking for?

**Comment by kubukoz** [comment]
yes, I looked at the class, and it's useful as a way of reading the build files, but one still has to implement the loading of such a build config into a model - that's the missing part (as far as I can tell).

so, in pseudocode:

```java
Path file = ...;
SmithyBuildConfig bc = SmithyBuildConfig.load(file);

// we don't have this
Model model = Model.fromBuildConfig(bc).assemble().unwrap();
// alternative syntax
Model model = bc.load().assemble().unwrap();
// another one
Model model = SmithyBuildConfig.load(bc).assemble().unwrap()
// also, the methods currently are shown to return ModelAssembler, but maybe it makes sense to return a ValidatedResult<Model> instead.
```

**Comment by kstich** [comment]
This is technically doable with something like the following:

```
SmithyBuildConfig.load(bc)
    .toModelAssembler(<optional classpath>)
    .assemble()
    .unwrap()
```

The problem comes in where a lot of the context in a `SmithyBuildConfig` is dropped, projections (including transforms) aren't carried through to `ModelAssembler`. We could make the method projection aware as `.toModelAssembler(String projectionName, <optional classpath>)`, and duplicate a bunch of work to ignore plugin execution. Or take a `projectionName` a different way and extract results from `SmithyBuild`

```
ProjectionResult projectionResult = new SmithyBuild(SmithyBuildConfig.load(bc))
        .build()
        .getProjectionResult(projectionName).orElseThrow();
return new ValidatedResult(projectionResult.getModel(), projectionResult.getEvents());
```

Which is in the box but harder to construct and does all the work for just the model.
